forge-game/src/main/java/forge/game/card/Card.java

```diff
@@ -2824,6 +2836,26 @@ else if (o.grantsZonePermissions())
                 sAbility = sbSA.toString();
             } else if (sa.isSpell() && sa.isBasicSpell()) {
                 continue;
+            } else if (sa.hasParam("DescriptionFromChosenName") && !getNamedCard().isEmpty()) {
+                String name = getNamedCard();
+                Card namedCard = Card.fromPaperCard(StaticData.instance().getCommonCards().getUniqueByName(name), getController());
+                CardState namedState = namedCard.getStateFromCardName(name);
```

Comment:
this seems unneeded
`getFaceByName` should provide what you want directly?

Reply:
Aaah, I knew something like that had to exist. Replaced it with that, and cleaned up the getStateFromCardName method I added because I kept needing it without realizing this was an option.